### PR TITLE
PaLM Enwiki8 example

### PR DESCRIPTION
**Pull request by conceptofmind**
Dear Dr. Wang,

I have put together an enwiki8 training example following a similar structure to the PyTorch training example. I attempted to adapt some of the code from your MLP-GPT JAX repository, https://github.com/lucidrains/mlp-gpt-jax, to use for the example.

Please ensure everything is up to your standards and running optimally. I tested to see if training loss was decreasing normally and samples were being generated at certain steps but I am not sure whether I adapted your code in HAIKU 100% properly.

I had contacted Patrick Kidger of Equinox to try to make sure code quality was at least adequate and functioning: patrick-kidger/equinox#95

If there are any noticeable issues I will work on resolving them.

Additionally, I am working on full pre-training example scripts for PaLM-pytorch, PaLM-jax, Flamingo-pytorch, and CoCa-pytorch utilizing HuggingFace streaming datasets and tokenizers. I would like to make further pull requests if each of the pre-training scripts meets your standards. I seem to see a lot of people asking for pre-training examples in the comments and issue tabs of the repositories, and thought it may be beneficial to their learning/understanding. I will try to replicate each of the pre-training examples in a similar fashion to the one provided in: https://github.com/lucidrains/reformer-pytorch/blob/master/pretraining/self-supervised.py

Also, I am still not able to upload enwiki8 to the data folder for some reason...

Thank you again,

Enrico

### CODE REVIEW

**Comment by lucidrains** [comment]
@conceptofmind thank you for this Enrico! looks great :100: :rocket:
